source/class/qx/io/jsonrpc/Client.js

```diff
@@ -208,8 +208,17 @@ qx.Class.define("qx.io.jsonrpc.Client", {
         params
       );
 
-      await this.send(request);
-      return await request.getPromise();
+      // sending request asychnonously without waiting for it completing, 
+      // since we're awaiting the fulfillment of the request promise below
+      this.send(request);
```

Comment:
This change appears to be simply masking the problem, not fixing it. `this.send()` is, in fact, an async call, so not awaiting it (or .then/.catch it) is simply hiding the rejection that was previously trapped by the unhandled rejection handler. No?

Reply:
This might be the case but it is not relevant because what we're waiting for is the fulfilment/rejection of the request's promise, not the promise of the `send()` method - which does not have any data in any case. The promise comes from the transport implementation's `send()` method; however, you are right that this masks an error that is raised by this method. I add a catch for that.

Reply:
The problem is that this "double await" - awaiting the transport to finish sending out the request and then awaiting thre request's promise to fulfil, seems to be creating the "unhandled promise rejection", since they seem to get in the way of each other. And we really do not need to await the completion of `send()`, we just need to handle if it rejects by rejecting the requests promise with a "Transport Error".

Reply:
It don't think this can happen - depending on the transport implementation the `send()` promise either fullfills when the request goes out and the answer has come in (XHR, fetch), and only afterwards is the request's promise settled, or it fulfills when the request has gone out (e.g. when using the PostMessage transport when communicating with a WeWorker). In any case, this promise is rejected when anything goes wrong during transport. The request's promise is settled afterwards in every case.

Reply:
we just need to handle `send()`-Promise rejections, I am just working on that.

Reply:
I now implemented it so the transport promise can be awaited as well, and using an environment variable to keep the old behaviour fully unless configured otherwise - thanks for the suggestions. I haven't updated the tests fully so they don't pass with the future behaviour, but will do so when I get to it.